src/query/v2/requests.hpp

```diff
@@ -458,21 +459,30 @@ struct ExpandOneResultRow {
   // The drawback of this is currently the key of the map is always interpreted as a string in Value, not as an
   // integer, which should be in case of mapped properties.
   Vertex src_vertex;
-  std::optional<Values> src_vertex_properties;
-  Values edges;
+  std::optional<std::map<PropertyId, Value>> src_vertex_properties;
+
+  // NOTE: If the desired edges are specified in the request,
+  // edges_with_specific_properties will have a value and it will
+  // return the properties as a vector of property values. The order
+  // of the values returned should be the same as the PropertyIds
+  // were defined in the request.
+  std::optional<std::vector<std::tuple<VertexId, Gid, std::map<PropertyId, Value>>>> edges_with_all_properties;
+  std::optional<std::vector<std::tuple<VertexId, Gid, std::vector<Value>>>> edges_with_specific_properties;
```

Comment:
I assume this was not a concern when the message type was defined. I am not sure how much gain in performance that would yield to us. This is a struct what is representing the request for one source vertex in the `ExpandOne` request, I think the reasoning behind the two possible forms of returning edge properties was because it is possible that one vertex would have a lot of connecting edges.

My opinion about changing the interface at this stage is the same as above,
but again, if you insist I will of course change this.